File: src/features/settings/domain/backup.ts
// src/features/settings/domain/backup.ts

export interface WebDavFormValues {
  webdavUrl: string;
  username: string;
  password?: string; // 仅在输入时存在，不回显
}

export interface CloudBackupFile {
  name: string;
  date: string;
  size: string;
}

// 定义整个 UI 的状态形状
export interface BackupState {
  isTesting: boolean;
  isBackingUp: boolean;
  isLoadingList: boolean;
  isDeleting: boolean;
  isConfigured: boolean;
  
  // 弹窗状态
  historyOpen: boolean;
  confirmOpen: boolean;
  
  // 数据
  backupList: CloudBackupFile[];
  selectedBackup: CloudBackupFile | null;
  //[新增] 本地操作状态
  isExporting: boolean;
  isImporting: boolean;
  //[新增] 进度条相关的状态
  progressValue: number;
  progressText: string;
  isProgressVisible: boolean;
  restorePreview: RestorePreview | null;
}
  //[新增] 对应后端的结构
export interface BackupMetadata {
  version: string;
  deviceId: string;
  deviceName: string;
  timestamp: number;
  platform: string;
}

export interface RestorePreview {
  tempFilePath: string;
  metadata?: BackupMetadata;
}

File: src-tauri/src/models/backup.rs
use serde::{Deserialize, Serialize};

pub type CommandResult<T> = Result<T, String>;

// [新增] 恢复预览信息 (下载后返回给前端)
#[derive(Debug, Serialize, Deserialize)] // 🟢 建议加上 Deserialize，万一以后要传回后端
#[serde(rename_all = "camelCase")]
pub struct RestorePreview {
    pub temp_file_path: String,
    pub metadata: Option<BackupMetadata>,
}

// 🟢 [修改] 同样添加 camelCase
#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct BackupMetadata {
    pub version: String,
    pub device_id: String,
    pub device_name: String,
    pub timestamp: i64,
    pub platform: String,
}

// 🟢 [优化] 加上 camelCase 以防万一，保持统一
#[derive(Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ProgressPayload {
    pub message: String,
    pub progress: f64,
}

// 🟢 [优化] 加上 Deserialize 和 camelCase
// 这样前端传回文件对象给后端时才不会报错
#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct CloudBackupFile {
    pub name: String,
    pub date: String,
    pub size: String,
}
